Update payment validation to use payments feature toggle config

aliveni opened the pull request:
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-813

### Change description ###
Update Payments validation to use the new payments config added in #845 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

doncem commented on src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java:
```diff
@@ -298,6 +304,10 @@ private ZipFileProcessingResult processZipFileContent(
             result.setEnvelope(envelopeProcessor.saveEnvelope(dbEnvelope));
 
             return result;
+        } catch (InvalidJourneyClassificationException ex) {
```
`InvalidJourneyClassificationException` no longer needs to be `InvalidEnvelopeException` extension ;)

doncem commented on src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java:
```diff
@@ -236,7 +236,8 @@ private static boolean isPaymentsEnabledForContainer(
             .findFirst()
-            .map(ContainerMappings.Mapping::isPaymentsEnabled).get();
+            .map(ContainerMappings.Mapping::isPaymentsEnabled)
+            .orElse(false);
```
👏

nitinprabhu commented on src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java:
```diff
@@ -207,16 +207,39 @@ public static void assertContainerMatchesJurisdictionAndPoBox(
-        if (envelope.payments != null && !envelope.payments.isEmpty()
-            && envelope.classification != Classification.NEW_APPLICATION) {
+    public static void assertPaymentsEnabledAndAllowedForClassification(
+        InputEnvelope envelope,
```
Could we not split these validations ? It would be much easier

aliveni replied:
done

nitinprabhu commented on src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java:
```diff
@@ -207,16 +207,39 @@ public static void assertContainerMatchesJurisdictionAndPoBox(
+            || !isPaymentsEnabledForContainer(mappings, envelope)
+            || envelope.classification != Classification.NEW_APPLICATION
+            )) {
             throw new InvalidJourneyClassificationException(
```
This looks incorrect exception ?

aliveni replied:
Added a new Exception for payments `PaymentsDisabledException`

aliveni commented:
> Please see comments

@nitinprabhu Addressed review comments. please review

nitinprabhu commented on src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java:
```diff
@@ -217,6 +218,39 @@ public static void assertClassificationNewApplicationIfPaymentsArePresent(InputE
+        boolean paymentsEnabled,
+        List<ContainerMappings.Mapping> mappings
+    ) {
+        if (envelope.payments != null && !envelope.payments.isEmpty()
```
you can just use `CollectionUtils.isNotEmpty(listName)`?

aliveni replied:
done

aliveni commented:
> Can we please add assertion in integration tests that when payment processing is disabled or disabled per container entry in notification table is present?
> 
> Pr looks good otherwise

Added integration test. Error mapping was missing for Payments exception, added that too
